File: 17-missing-rust/samples/pi.pl
use strict;
use warnings;
my ($n, $d) = (355, 113);
my $whole = int($n / $d);
my $rem = $n % $d;
my $digits = '';
for (1 .. 4) {
  $rem *= 10;
  $digits .= int($rem / $d);
  $rem %= $d;
}
print "hello $whole.$digits world!\n";

File: 17-missing-rust/samples/pi.f90
program pi_probe
  implicit none
  integer :: n, d, whole, rem, i
  character(len=4) :: digits

  n = 355; d = 113
  whole = n / d; rem = mod(n, d)
  do i = 1, 4
    rem = rem * 10
    write(digits(i:i), '(I1)') rem / d
    rem = mod(rem, d)
  end do
  write (*, '(A,I1,A,A,A)') 'hello ', whole, '.', digits, ' world!'
end program pi_probe
